src/poetry/console/commands/installer_command.py

```diff
@@ -30,3 +33,7 @@ def installer(self) -> Installer:
 
     def set_installer(self, installer: Installer) -> None:
         self._installer = installer
+
+    def execute(self, io: IO) -> int:
+        PoetryKeyring.preflight_check(io)
```

Comment:
```suggestion
        PoetryKeyring.preflight_check(io, self.poetry.config)
```

Otherwise, we are neglecting the local config IIRC.